concept/impl/ConceptManagerImpl.java

```diff
@@ -377,18 +378,20 @@ Attribute getCachedAttribute(String index) {
 
     /**
      * This is only used when checking if attribute exists before trying to create a new one.
-     * We use a readLock as janusGraph commit does not seem to be atomic. Further investigation needed
      */
-    Attribute getAttributeWithLock(String index) {
+    Attribute getAttribute(String index) {
         Attribute concept = getCachedAttribute(index);
         if (concept != null) return concept;
 
-        graphLock.readLock().lock();
-        try {
-            return getConcept(Schema.VertexProperty.INDEX, index);
-        } finally {
-            graphLock.readLock().unlock();
-        }
+        //check EPHA
+        if (attributeManager.attributesEphemeral().containsKey(index)) return null;
```

Comment:
if not in ephemeral it can still be in the graph 🌞

Reply:
yes, that's why we keep checking